lib/Importer/Caffe2ModelLoader.cpp

```diff
@@ -623,9 +618,9 @@ llvm::Error Caffe2ModelLoader::loadOperator(const caffe2::OperatorDef &op) {
       auto outTy = G_.getParent()->uniqueType(
           ElemKind::Int8QTy, {in.getType()->dims()[0], B->getType()->dims()[0]},
           yScale, yZeroPoint - OFFSETSHIFT);
-      node = G_.createFullyConnected(opName, in, W, B, outTy);
+      node = G_.createFullyConnected(opName, in, W, B, outTy, axis);
     } else {
-      node = G_.createFullyConnected(opName, in, W, B);
+      node = G_.createFullyConnected(opName, in, W, B, axis);
```

Comment:
Should we move the stuff below starting at line 626 as well?

Reply:
Good question -- the reason I didn't was that I was thinking of the reshaping back after the FC as more specific to Caffe2's FC semantics. Meanwhile I think of requiring FC inputs to be 2D as independent of Caffe2/ONNX/etc., so that's why it made sense to move the reshape to the creator and add an IR constraint in the verification. What do you think?

Reply:
Hm yeah that kinda makes sense to me too